zrml/prediction-markets/src/lib.rs

```diff
@@ -232,26 +234,61 @@ mod pallet {
             categories: u16,
         ) -> DispatchResult {
             let sender = ensure_signed(origin)?;
+            Self::ensure_create_market_end(end)?;
 
             ensure!(
                 categories <= T::MaxCategories::get(),
                 "Cannot exceed max categories for a new market."
             );
 
-            match end {
-                MarketEnd::Block(block) => {
-                    let current_block = <frame_system::Pallet<T>>::block_number();
-                    ensure!(current_block < block, Error::<T>::EndBlockTooSoon);
+            let status: MarketStatus = match creation {
+                MarketCreation::Permissionless => {
+                    let required_bond = T::ValidityBond::get() + T::OracleBond::get();
+                    T::Currency::reserve(&sender, required_bond)?;
+                    MarketStatus::Active
                 }
-                MarketEnd::Timestamp(timestamp) => {
-                    let now = <pallet_timestamp::Pallet<T>>::get();
-                    ensure!(
-                        now < timestamp.saturated_into(),
-                        Error::<T>::EndTimestampTooSoon
-                    );
+                MarketCreation::Advised => {
+                    let required_bond = T::AdvisoryBond::get() + T::OracleBond::get();
+                    T::Currency::reserve(&sender, required_bond)?;
+                    MarketStatus::Proposed
                 }
             };
 
+            let market_id = Self::get_next_market_id()?;
+            let market = Market {
+                creator: sender.clone(),
+                creation,
+                creator_fee: 0,
+                oracle,
+                end,
+                metadata,
+                market_type: MarketType::Categorical(categories),
+                status,
+                report: None,
+                resolved_outcome: None,
+            };
+
+            <Markets<T>>::insert(market_id.clone(), Some(market));
+
+            Self::deposit_event(Event::MarketCreated(market_id, sender));
+
+            Ok(())
+        }
+
+        #[pallet::weight(10_000)]
```

Comment:
Weight is too low. For example, all functions we have so far in the Swaps pallet have a lower weight bound of 50 million (excl. db interactions)